public/schemasystem/schematypes.h

```diff
@@ -0,0 +1,427 @@
+	SchemaTypeCategory_t m_eTypeCategory;
+	SchemaAtomicCategory_t m_eAtomicCategory;
+};
+
+class CSchemaType_Builtin : public CSchemaType
+{
+public:
+	SchemaBuiltinType_t m_eBuiltinType;
+	uint8 m_nSize;
+};
+
+class CSchemaType_Ptr : public CSchemaType
+{
+public:
+	CSchemaType* m_pObjectType;
+};
+
+class CSchemaType_Atomic : public CSchemaType
+{
+public:
+	SchemaAtomicTypeInfo_t* m_pAtomicInfo;
+	int m_nAtomicID;
+	uint16 m_nSize;
+	uint8 m_nAlignment;
+};
+
+class CSchemaType_Atomic_T : public CSchemaType_Atomic
+{
+public:
+	CSchemaType* m_pTemplateType;
+};
+
+class CSchemaType_Atomic_CollectionOfT : public CSchemaType_Atomic_T
+{
+public:
+	SchemaAtomicTHandlerFn_t m_pfnHandler;
+	uint16 m_nElementSize;
+};
+
+class CSchemaType_Atomic_TF : public CSchemaType_Atomic_T
+{
+public:
+	int m_nFuncPtrSize;
+};
+
+class CSchemaType_Atomic_TT : public CSchemaType_Atomic_T
+{
+public:
+	CSchemaType* m_pTemplateType2;
+};
+
+class CSchemaType_Atomic_TTF : public CSchemaType_Atomic_TT
+{
+public:
+	int m_nFuncPtrSize;
+};
+
+class CSchemaType_Atomic_I : public CSchemaType_Atomic
+{
+public:
+	int m_nInteger;
+};
+
+class CSchemaType_DeclaredClass : public CSchemaType
+{
+public:
+	CSchemaClassInfo* m_pClassInfo;
+	bool m_bGlobalPromotionRequired;
+};
+
+class CSchemaType_DeclaredEnum : public CSchemaType
+{
+public:
+	CSchemaEnumInfo* m_pEnumInfo;
+	bool m_bGlobalPromotionRequired;
+};
+
+class CSchemaType_FixedArray : public CSchemaType
+{
+public:
+	int m_nElementCount;
+	uint16 m_nElementSize;
+	uint8 m_nElementAlignment;
+	CSchemaType* m_pElementType;
+};
+
+class CSchemaType_Bitfield : public CSchemaType
+{
+public:
+	int m_nSize;
+};
+
+struct SchemaMetadataEntryData_t
+{
+	const char* m_pszName;
+	void* m_pData;
+};
+
+struct SchemaClassFieldData_t
+{
+	const char* m_pszName;
+	
+	CSchemaType* m_pType;
+	
+	int m_nSingleInheritanceOffset;
+	
+	int m_nStaticMetadataCount;
+	SchemaMetadataEntryData_t* m_pStaticMetadata;
+};
+
+struct SchemaStaticFieldData_t
+{
+	const char* m_pszName;
+	
+	CSchemaType* m_pType;
+	
+	void* m_pInstance;
+	
+	int m_nStaticMetadataCount;
+	SchemaMetadataEntryData_t* m_pStaticMetadata;
+};
+
+struct SchemaBaseClassInfoData_t
+{
+	uint m_nOffset;
+	CSchemaClassInfo* m_pClass;
+};
+
+struct SchemaClassInfoData_t
+{
+	CSchemaClassInfo* m_pSchemaBinding;
+	
+	const char* m_pszName;
+	const char* m_pszProjectName;
+	
+	int m_nSize;
+	
+	uint16 m_nFieldCount;
+	uint16 m_nStaticFieldCount;
+	uint16 m_nStaticMetadataCount;
+	
+	uint8 m_nAlignment;
+	uint8 m_nBaseClassCount;
+	
+	uint16 m_nMultipleInheritanceDepth;
+	uint16 m_nSingleInheritanceDepth;
+	
+	SchemaClassFieldData_t* m_pFields;
+    SchemaStaticFieldData_t* m_pStaticFields;
+    SchemaBaseClassInfoData_t* m_pBaseClasses;
+    datamap_t* m_pDataDescMap;
+    SchemaMetadataEntryData_t* m_pStaticMetadata;
+	
+    CSchemaSystemTypeScope* m_pTypeScope;
+    CSchemaType_DeclaredClass* m_pDeclaredClass;
+	
+    uint m_nFlags1;
+    uint m_nFlags2;
```

Comment:
Could this be a single uint64 instead? That would make more sense imo

Reply:
It looks like the game dereferences the first half as int32.
![Безымянный11111](https://github.com/alliedmodders/hl2sdk/assets/152704532/d9d92bae-8176-42ab-a380-25b487281815)
I'm not entirely sure that m_nFlags2 are really flags, since I haven't seen its use anywhere. My guess about their separation is based on the 'CF1' in the name of the flags.

Reply:
Hm, well that's a strange separation, especially if the 2nd portion isn't used, but alright